### PR TITLE
Update DiskThresholdDeciderIT to verify what shard was actually allocated

### PR DESCRIPTION
**Pull request by idegtiarenko**
While working on desired-balance branch it was discovered that `DiskThresholdDeciderIT#testRestoreSnapshotAllocationDoesNotExceedWatermark` might fail very rarely. 

According to the investigation this could happen if `SnapshotShardSizeInfo` does not contain one of the shards and balance is wrongly calculated with 0 size assumption for a missing shard. Later on it could not be reconciled and a small node is left with 0 shards failing the test scenario.

On the main branch the test seems to be stable, however there is no verification that a **correct** shard is assigned to the smallest node leaving a space for a latent bug.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java [hunk 1]
```diff
@@ -88,16 +91,16 @@ public void testHighWatermarkNotExceeded() throws Exception {
...
         // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
-        getTestFileStore(dataNodeName).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
+        getTestFileStore(dataNodeName).setTotalSpace(smallestShard.getValue() + WATERMARK_BYTES - 1L);
         assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, empty());
 
         // increase disk size of node 0 to allow just enough room for one shard, and check that it's rebalanced back
-        getTestFileStore(dataNodeName).setTotalSpace(minShardSize + WATERMARK_BYTES + 1L);
-        assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, hasSize(1));
+        getTestFileStore(dataNodeName).setTotalSpace(smallestShard.getValue() + WATERMARK_BYTES + 1L);
+        assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, contains(smallestShard.getKey()));
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java:103]: I think this is too strong - we might have two shards of the same (smallest) size and in that case we might choose to assign either.

Comment (idegtiarenko) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java]: Good point. Updated.

**Review by DaveCTurner — COMMENTED** [review]
Change LGTM but I think this is flaky right? If so, we shouldn't merge it without fixing that flakiness.

**Review by DaveCTurner — APPROVED** [review]
We discussed in another channel and think this is now fixed, so LGTM

**Pull request merged**